public/app/plugins/datasource/graphite/func_editor.ts

```diff
@@ -0,0 +1,318 @@
+
+          var paramValue = templateSrv.highlightVariablesAsHtml(func.params[index]);
+
+          var last = index >= func.params.length - 1 && param.optional && !paramValue;
+          if (last && param.multiple) {
+            paramValue = '+';
+          }
+
+          if (index > 0) {
+            $('<span class="comma' + (last ? ' query-part__last' : '') + '">, </span>').appendTo(elem);
+          }
+
+          var $paramLink = $(
+            '<a ng-click="" class="graphite-func-param-link' +
+              (last ? ' query-part__last' : '') +
+              '">' +
+              (paramValue || '&nbsp;') +
+              '</a>'
+          );
+          var $input = $(paramTemplate);
+          $input.attr('placeholder', param.name);
+
+          paramCountAtLink++;
+
+          $paramLink.appendTo(elem);
+          $input.appendTo(elem);
+
+          $input.blur(_.partial(inputBlur, index));
+          $input.keyup(inputKeyDown);
+          $input.keypress(_.partial(inputKeyPress, index));
+          $paramLink.click(_.partial(clickFuncParam, index));
+
+          if (param.options) {
+            addTypeahead($input, index);
+          }
+
+          return true;
+        });
+
+        $('<span>)</span>').appendTo(elem);
+
+        $compile(elem.contents())($scope);
+      }
+
+      function ifJustAddedFocusFirstParam() {
+        if ($scope.func.added) {
+          $scope.func.added = false;
+          setTimeout(function() {
+            elem
+              .find('.graphite-func-param-link')
+              .first()
+              .click();
+          }, 10);
+        }
+      }
+
+      function registerFuncControlsToggle() {
+        $funcLink.click(toggleFuncControls);
+      }
+
+      function registerFuncControlsActions() {
+        $funcControls.click(function(e) {
+          var $target = $(e.target);
+          if ($target.hasClass('fa-remove')) {
+            toggleFuncControls();
+            $scope.$apply(function() {
+              ctrl.removeFunction($scope.func);
+            });
+            return;
+          }
+
+          if ($target.hasClass('fa-arrow-left')) {
+            $scope.$apply(function() {
+              _.move(ctrl.queryModel.functions, $scope.$index, $scope.$index - 1);
+              ctrl.targetChanged();
+            });
+            return;
+          }
+
+          if ($target.hasClass('fa-arrow-right')) {
+            $scope.$apply(function() {
+              _.move(ctrl.queryModel.functions, $scope.$index, $scope.$index + 1);
+              ctrl.targetChanged();
+            });
+            return;
+          }
+
+          if ($target.hasClass('fa-question-circle')) {
+            var funcDef = ctrl.datasource.getFuncDef(func.def.name);
+            if (funcDef && funcDef.description) {
+              popoverSrv.show({
+                element: e.target,
+                position: 'bottom left',
+                classNames: 'drop-popover drop-function-def',
+                template:
```

Comment:
Should be able to use string interpolation here, i.e. `` `string text ${expression} string text` ``, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals